src/main/scala/viper/gobra/frontend/Parser.scala

```diff
@@ -86,8 +95,37 @@ object Parser {
       }
     }
 
+    def parseSourceCached(source: FromFileSource): Either[Vector[VerifierError], PProgram] = {
+      var cacheHit = true
+      def parseAndStore(): (Either[Vector[VerifierError], PProgram], Positions) = {
+        cacheHit = false
+        val res = parseSource(source)
+        sourceCache += (source.path.toString, source.content) -> (res, positions)
```

Comment:
Having the entire content of the program in the Map key might cause unnecessary slowdowns in the cache lookup and may cause unnecessary memory consumption (for big projects). Could we instead store an hash of the AST?

Reply:
I'm taking now the MD5 hash of the path and content as the key